### PR TITLE
PACA-810: DW-Micrometer conversion for RDA pipeline

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-810](https://jira.cms.gov/browse/PACA-810)

**User Story or Bug Summary:**

As a PACA engineer I want to generate CloudWatch alerts and dashboards on RDA API pipeline job metrics.

---

### What Does This PR Do?

This PR is intended for testing in the TEST environment.  It includes the following changes:

- Add Micrometer library to the pipeline app.
- Convert all RDA API pipeline components to track metrics using Micrometer instead of DropWizard.
- Ensure all Micrometer metrics are also published transparently to NewRelic through DropWizard.
- Add support for sending metrics to CloudWatch if `MICROMETER_CW_ENABLED` environment variable is set to `true`.
- Add support for sending metrics to JMX if `MICROMETER_JMX_ENABLED` environment variable is set to `true`.
- Add support for configuring various Micrometer settings using environment variables (see below).

| Env Var Name | Default | Description |
|---|---|---|
| MICROMETER_CW_ENABLED | false | Enables transfer of metrics to CW when `true` |
| MICROMETER_CW_NAMESPACE |  | Sets the namespace to use when sending metrics to CW.   Has no default.  Required if enabled is true. |
| MICROMETER_CW_INTERVAL | PT1M | [ISO-8601 time interval](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/Duration.html#parse(java.lang.CharSequence)) specifying how often to send metrics to CW. |
| MICROMETER_JMX_ENABLED | false | Enables transfer of metrics to local JMX when enabled. |

**New Dependencies added in this PR**

These dependencies add the Micrometer libraries that allow for tracking metric values and reporting them to JMX, CloudWatch, and/or NewRelic.  Currently these are only being used in the ETL pipeline app and related subprojects.

```
<micrometer.version>1.10.0</micrometer.version>
<dependency>
	<!-- Used to collect metrics in a registry that can be transmitted to CloudWatch efficiently. -->
	<groupId>io.micrometer</groupId>
	<artifactId>micrometer-core</artifactId>
	<version>${micrometer.version}</version>
</dependency>
<dependency>
	<!-- Used to transmit metrics to JMX for local testing. -->
	<groupId>io.micrometer</groupId>
	<artifactId>micrometer-registry-jmx</artifactId>
	<version>${micrometer.version}</version>
</dependency>
<dependency>
	<!-- Used to transmit metrics to CloudWatch for alerts and dashboards. -->
	<groupId>io.micrometer</groupId>
	<artifactId>micrometer-registry-cloudwatch</artifactId>
	<version>${micrometer.version}</version>
</dependency>
<dependency>
	<!-- Used to transmit metrics to NewRelic for alerts and dashboards. -->
	<groupId>io.micrometer</groupId>
	<artifactId>micrometer-registry-newrelic</artifactId>
	<version>${micrometer.version}</version>
</dependency>
```

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [x] Yes
      * [ ] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [x] Yes
      * [ ] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSinkTest.java [hunk 1]
```diff
@@ -185,25 +186,25 @@ public void testSingleMessageTransformFailureUpdatesMetric() {
...
-    assertEquals(List.of(0L, 100L), getHistogramValues(sink.getMetrics().getChangeAgeMillis()));
-    assertEquals(List.of(0L, 250L), getHistogramValues(sink.getMetrics().getExtractAgeMillis()));
+    assertEquals(2, sink.getMetrics().getChangeAgeMillis().count());
+    assertEquals(2, sink.getMetrics().getExtractAgeMillis().count());
+    assertEquals(50, sink.getMetrics().getChangeAgeMillis().mean());
+    assertEquals(125, sink.getMetrics().getExtractAgeMillis().mean());
+    //    assertEquals(List.of(0L, 100L),
+    // getHistogramValues(sink.getMetrics().getChangeAgeMillis()));
+    //    assertEquals(List.of(0L, 250L),
+    // getHistogramValues(sink.getMetrics().getExtractAgeMillis()));
```
Comment (KyRobbins) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSinkTest.java:205]: Old code cleanup

Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSinkTest.java]: Fixed.  Thanks.

**Review by KyRobbins — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSourceTest.java [hunk 2]
```diff
@@ -86,8 +87,8 @@ public class StandardGrpcRdaSourceTest {
   /** String used as a RDA API "version" in the unit tests. */
   public static final String VERSION = "version";
 
-  /** A MetricRegistry used to verify metrics. */
-  private MetricRegistry appMetrics;
+  /** A MeterRegistry used to verify metrics. */
```
Comment (mackec1) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/StandardGrpcRdaSourceTest.java:90]: minor nit; we've been trying to use things like {@link MetricRegistry} in JavaDoc(s)

**Review by mackec1 — APPROVED** [review]
Looks really nice; hopeful BFD can leverage that work to do same in Pipeline. Thx.

File: apps/pom.xml [hunk 3]
```diff
@@ -200,6 +201,31 @@
...
+				<groupId>io.micrometer</groupId>
+				<artifactId>micrometer-registry-cloudwatch</artifactId>
+				<version>${micrometer.version}</version>
+			</dependency>
+			<dependency>
+				<!-- Used to transmit metrics to NewRelic for alerts and dashboards. -->
+				<groupId>io.micrometer</groupId>
+				<artifactId>micrometer-registry-newrelic</artifactId>
+				<version>${micrometer.version}</version>
+			</dependency>
```
Comment (mjburling) [inline apps/pom.xml:227]: 👋 Hey, @StewGoin can you provide any input on including this suite of micrometer libraries? Thanks!

Comment (StewGoin) [reply apps/pom.xml:227]: No concerns from me on using this here as well.

**Review by StewGoin — APPROVED** [review]
Ok by me (ISSO) to proceed.

**Pull request merged**